internal/backend/sftp/sftp.go

```diff
@@ -181,10 +183,10 @@ func (r *SFTP) mkdirAllDataSubdirs(ctx context.Context, nconn uint) error {
 			// round trip, not counting duplicate parent creations causes by
 			// concurrency. MkdirAll first does Stat, then recursive MkdirAll
 			// on the parent, so calls typically take three round trips.
-			if err := r.c.Mkdir(d); err == nil {
+			if err := r.Mkdir(d); err == nil {
```

Comment:
The idea of the old implementation is to speculatively create the directory and only resort to more expensive code paths if that didn't work. After all the default situation here is that the parent folder already exists and only the new child folder has to be created. So this PR shouldn't break that optimization.